test/bats/vault.bats

```diff
@@ -11,15 +11,25 @@ export ANNOTATION_VALUE=${ANNOTATION_VALUE:-"app=test"}
 
 @test "install vault provider" {
   # create the ns vault
+
+  oc delete project vault --ignore-not-found=true
   kubectl create ns vault
+  kubectl label ns vault security.openshift.io/scc.podSecurityLabelSync=false pod-security.kubernetes.io/enforce=privileged pod-security.kubernetes.io/audit=privileged pod-security.kubernetes.io/warn=privileged --overwrite
+  oc adm policy add-scc-to-user privileged system:serviceaccount:vault:vault
+  oc adm policy add-scc-to-user privileged system:serviceaccount:vault:vault-csi-provider
+
   # install the vault provider using the helm charts
   # pinning this to a fixed version (1.7.0)
   helm repo add hashicorp https://helm.releases.hashicorp.com
   helm repo update
   helm install vault hashicorp/vault --namespace=vault \
         --set "server.dev.enabled=true" \
         --set "injector.enabled=false" \
-        --set "csi.enabled=true"
+        --set "csi.enabled=true" \
+        --set "global.openshift=true" \
+        --set "csi.daemonSet.providersDir=/var/run/secrets-store-csi-providers"
```

Comment:
I tried to run the script in my cluster and it tried to pull vault images from `registry.redhat.io`. I had to add:

```
--set "injector.agentImage.repository=docker.io/hashicorp/vault"  \
--set "server.image.repository=docker.io/hashicorp/vault" \
--set "csi.image.repository=docker.io/hashicorp/vault-csi-provider" \
--set "csi.agent.image.repository=docker.io/hashicorp/vault"
```

Is that expected? Did I miss any configuration?

Reply:
1. git clone git@github.com:ropatil010/secrets-store-csi-driver.git -b hackvault_create hackvault_create
2. Install secrets csi operator, driver
3. cd hackvault_create
4. make e2e-vault
oc get pods -n vault
NAME                       READY   STATUS    RESTARTS   AGE
vault-0                    1/1     Running   0          77s
vault-csi-provider-9bnn6   2/2     Running   0          65s
vault-csi-provider-dnf89   2/2     Running   0          61s
vault-csi-provider-vcdf5   2/2     Running   0          60s

rohitpatil@ropatil-mac hackvault_create % make e2e-vault
bats -t test/bats/vault.bats
1..17
ok 1 install vault provider
ok 2 setup vault
ok 3 deploy vault secretproviderclass crd
...
ok 16 deploy pod with multiple secret provider class
ok 17 CSI inline volume test with multiple secret provider class